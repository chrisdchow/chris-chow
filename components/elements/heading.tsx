import { FunctionComponent } from 'react';

type HeadingProps = Record<string, unknown>;

/**
 * https://tailwindcss.com/docs/preflight
 * Headings are unstyled (normalized)
 */

const baseHeadingStyle = 'font-bold leading-loose my-4 mx-0';

export const H1: FunctionComponent<HeadingProps> = ({ children }) => {
  return <h1 className={`${baseHeadingStyle} text-3xl`}>{children}</h1>;
};

export const H2: FunctionComponent<HeadingProps> = ({ children }) => {
  return <h2 className={`${baseHeadingStyle} text-2xl`}>{children}</h2>;
};

export const H3: FunctionComponent<HeadingProps> = ({ children }) => {
  return <h3 className={`${baseHeadingStyle} text-xl`}>{children}</h3>;
};

export const H4: FunctionComponent<HeadingProps> = ({ children }) => {
  return <h4 className={`${baseHeadingStyle} text-l`}>{children}</h4>;
};
