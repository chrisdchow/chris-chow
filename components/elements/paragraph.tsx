import { FunctionComponent } from 'react';

type ParagraphProps = Record<string, unknown>;

/**
 * https://tailwindcss.com/docs/preflight
 * Paragraph tags are unstyled (normalized)
 */

export const P: FunctionComponent<ParagraphProps> = ({ children }) => {
  return <p className='my-2 whitespace-pre-wrap'>{children}</p>;
};
