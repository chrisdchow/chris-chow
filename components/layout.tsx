import { FunctionComponent } from 'react';
import Head from 'next/head';
import Link from 'next/link';
import { H1, H2 } from '@components/elements/heading';

const name = 'Chris Chow';
export const siteTitle = name;

type LayoutProps = {
  home?: boolean;
};

export const Layout: FunctionComponent<LayoutProps> = ({ home, children }): JSX.Element => {
  const homeHeader = (
    <>
      <img src='/images/profile.jpg' className='h-32 w-32 rounded-full' alt={name} />
      <H1>{name}</H1>
    </>
  );

  const navigateHomeHeader = (
    <>
      <Link href='/'>
        <a>
          <img src='/images/profile.jpg' className='h-32 w-32 rounded-full' alt={name} />
        </a>
      </Link>
      <H2>
        <Link href='/'>
          <a>{name}</a>
        </Link>
      </H2>
    </>
  );

  return (
    <div className='max-w-xl py-0 px-4 mx-auto my-12'>
      <Head>
        <link rel='icon' href='/favicon.ico' />
        <meta name='description' content='Chris Chow' />
        <meta property='og:image' content={`https://og-image.now.sh/${encodeURI(siteTitle)}.png`} />
        <meta name='og:title' content={siteTitle} />
        <meta name='twitter:card' content='summary_large_image' />
      </Head>
      <header className='flex flex-col items-center'>{home ? homeHeader : navigateHomeHeader}</header>
      <main>{children}</main>
      {!home && (
        <div className='mt-12 mx-0 mb-0'>
          <Link href='/'>
            <a>‚Üê Back to home</a>
          </Link>
        </div>
      )}
    </div>
  );
};
