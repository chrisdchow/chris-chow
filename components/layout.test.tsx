import React from 'react';
import { render } from '@test/testUtils';
import { Layout } from './layout';

describe('Layout compnent', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Layout home={false} />, {});
    expect(asFragment()).toMatchSnapshot();
  });
});
