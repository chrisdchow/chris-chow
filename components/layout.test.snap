// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout compnent matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <header
      class="header"
    >
      <a
        href="/"
      >
        <img
          alt="Chris Chow"
          class="headerImage borderCircle"
          src="/images/profile.jpg"
        />
      </a>
      <h2
        class="headingLg"
      >
        <a
          class="colorInherit"
          href="/"
        >
          Chris Chow
        </a>
      </h2>
    </header>
    <main />
    <div
      class="backToHome"
    >
      <a
        href="/"
      >
        ‚Üê Back to home
      </a>
    </div>
  </div>
</DocumentFragment>
`;
